File: Chess-ios/Chess/asd/Spot.swift
//
//  Spot.swift
//  chess-program
//
//  Created by Cseri Ádám on 2022. 05. 03..
//

import Foundation

class Spot {
    let id: String
    var highlighted: Bool
    var piece: Piece
    
    init(piece: Piece = Empty() as Piece) {
        self.piece = piece
        self.highlighted = false
        self.id = String(Spot.setId)
        Spot.updateIdSetter()
    }
    
    static var setId = Array("a8")
    static func updateIdSetter() {
        if setId[0].inc() > "h" {
            setId[0] = "a"
            if setId[1].dec() < "1" {
                setId[1] = "8"
            }
        }
    }
}

File: Chess-android/app/src/main/kotlin/shared_code/Spot.kt
package shared_code

class Spot(_piece: Piece = Empty()) {
    var piece: Piece
    val id: String
    var highlighted: Boolean

    /**
     * Static-like variable and function. It gives each spot a common unique id in chess.
     * The id contains a file (a-h) and a rank (1-8).
     *
     * a8 b8 c8 ..
     * a7 b7 ..
     * a6 ..
     * ..
     */
    companion object {
        val setId = mutableListOf('a', '8')
        fun updateIdSetter() {
            if (++setId[0] > 'h') {
                setId[0] = 'a'
                if (--setId[1] < '1') {
                    setId[1] = '8'
                }
            }
        }
    }

    init {
        piece = _piece
        highlighted = false
        id = setId.joinToString("")
        updateIdSetter()
    }
}
